New Integration: Consul Connect (#9237)
DataDog/integrations-core: consul_connect/README.md

# Agent Check: Consul Connect

## Overview

> Monitor your [Consul Connect][1] Envoy sidecar proxies with the [Datadog Envoy Integration][2]. Although Consul Connect has support for external proxies, this setup refers to [Consul Connect configured with Envoy][3]. 

Reviewer: In the future would we document support for other proxies here?
```suggestion
Monitor your [Consul Connect][1] Envoy sidecar proxies with the [Datadog Envoy Integration][2]. The Consul Connect integration currently only supports [Consul Connect configured with Envoy][3]. 
```

## Setup

### Installation

Install the Datadog Agent on your services running Consul Connect and follow the [Configuration](#configuration) instructions for your appropriate environment.

### Configuration
Follow the instructions below to configure this check for an Agent running on a host. For containerized environments, see the [Containerized](#containerized) section.

<!-- xxx tabs xxx -->
<!-- xxx tab "Host" xxx -->

#### Host

To configure this check for an Agent running on a host:

##### Metric Collection

> Use config option [`-admin-bind`][4] in Consul Connect to ensure the Envoy Admin API is exposed, then follow the [Envoy host instructions][5] to configure metric collection.

Reviewer: Is the default sufficient or are you suppose to set `-admin-bind` to a different port?

# Agent Check: Consul Connect

## Overview

Monitor your [Consul Connect][1] Envoy sidecar proxies with the [Datadog Envoy Integration][2]. The Consul Connect integration currently only supports [Consul Connect configured with Envoy][3]. 

## Setup

### Installation

Install the Datadog Agent on your services running Consul Connect and follow the [Configuration](#configuration) instructions for your appropriate environment.

### Configuration
Follow the instructions below to configure this check for an Agent running on a host. For containerized environments, see the [Containerized](#containerized) section.

<!-- xxx tabs xxx -->
<!-- xxx tab "Host" xxx -->

#### Host

To configure this check for an Agent running on a host:

##### Metric Collection

> Follow the [Envoy host instructions][4] to configure metric collection.
> 
> **Note:** Use config option [`-admin-bind`][5] in Consul Connect to configure the port where the Envoy Admin API will be exposed.
> 

Reviewer: I think it's easier to understand if we follow a numbered instruction format.

```suggestion
1. In Consul Connect, enable the config option [`-admin-bind`][5] to configure the port where the Envoy Admin API will be exposed.

2. Enable the [Envoy integration][4] to configure metric collection.

```

##### Log Collection
Follow the [Envoy host][6] instructions to configure log collection.  

<!-- xxz tab xxx -->
<!-- xxx tab "Containerized" xxx -->

#### Containerized

> Follow the [Envoy containerized instructions][7] to configure your Datadog Agent for Envoy. 
> 
> ##### Metric collection
> Use config option [`envoy_stats_bind_addr`][8] in Consul Connect to ensure the `/stats` endpoint is exposed on the public network, then follow the [Envoy containerized instructions][9] to configure metric collection. 

Reviewer: ```suggestion
Follow the [Envoy containerized instructions][7] to configure your Datadog Agent for Envoy. 

##### Metric collection
1. In Consul Connect, enable the config option [`envoy_stats_bind_addr`][8] to ensure the `/stats` endpoint is exposed on the public network

 2. Configure the [Envoy integration for containerized environments instructions][9] to start collecting metrics. 
```
